walk-in-interview/src/main/java/com/google/job/data/JobPayment.java

```diff
@@ -61,24 +74,50 @@ public int hashCode() {
         c = paymentFrequency.hashCode();
         result = 31 * result + c;
 
+        c = ((Long) annualMax).hashCode();
+        result = 31 * result + c;
+
         this.hashCode = result;
 
         return hashCode;
     }
 
     @Override
     public String toString() {
-        return String.format("JobPayment{min=%d, max=%d, paymentFrequency=%s}",
-                min, max, paymentFrequency.name());
+        return String.format("JobPayment{min=%d, max=%d, paymentFrequency=%s, annualMax=%d}",
+                min, max, paymentFrequency.name(), annualMax);
     }
 
-    private static void validateParameter(float min, float max) throws IllegalArgumentException {
+    private static void validateParameter(float min, float max, PaymentFrequency paymentFrequency, long annualMax)
```

Comment:
the more i read those validateParameter methods you have everywhere the more i think it should be inlined in the constructor. separating the param validation from constructor forces you to pass all the arguments(so the method gets longer and longer) when it would be more readable to put this in the constructor directly at least you would know right there was is a valid param instead of deferring the knowledge and forcing someone to find that out by scrollin gall the way down. And if any validation is complicated and requires pre computation (beyond an if+throw) then the computation needed for the validation should be moved to a method. but i would recommend you inline this code instead.